analyses/org.eclipse.tracecompass.incubator.executioncomparision.ui/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/ui/ExecutionComparisonView.java

```diff
@@ -0,0 +1,1212 @@
+import org.eclipse.tracecompass.tmf.ui.viewers.ILegendImageProvider2;
+import org.eclipse.tracecompass.tmf.ui.viewers.TmfTimeViewer;
+import org.eclipse.tracecompass.tmf.ui.viewers.TmfViewer;
+import org.eclipse.tracecompass.tmf.ui.viewers.tree.AbstractSelectTreeViewer2;
+import org.eclipse.tracecompass.tmf.ui.viewers.tree.ICheckboxTreeViewerListener;
+import org.eclipse.tracecompass.tmf.ui.viewers.tree.ITmfTreeViewerEntry;
+import org.eclipse.tracecompass.tmf.ui.viewers.xychart.TmfXYChartViewer;
+import org.eclipse.tracecompass.tmf.ui.viewers.xychart.XYChartLegendImageProvider;
+import org.eclipse.tracecompass.tmf.ui.viewers.xychart.linechart.TmfCommonXAxisChartViewer;
+import org.eclipse.tracecompass.tmf.ui.viewers.xychart.linechart.TmfFilteredXYChartViewer;
+import org.eclipse.tracecompass.tmf.ui.viewers.xychart.linechart.TmfXYChartSettings;
+import org.eclipse.tracecompass.tmf.ui.views.ManyEntriesSelectedDialogPreCheckedListener;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.dialogs.TriStateFilteredCheckboxTree;
+import org.eclipse.ui.IWorkbenchPartSite;
+import org.eclipse.ui.contexts.IContextActivation;
+import org.eclipse.ui.contexts.IContextService;
+
+/**
+ * The ExecutionComparisonView allows to compare two groups of traces (parts of
+ * traces)
+ *
+ * @author Fateme Faraji Daneshgar
+ */
+@SuppressWarnings("restriction")
+public class ExecutionComparisonView extends DifferentialFlameGraphView implements ICheckboxTreeViewerListener {
+
+    /**
+     * the id of the view
+     */
+    public static final String VIEW_ID = "org.eclipse.tracecompass.incubator.internal.executioncomparison.ui.execComparison"; //$NON-NLS-1$
+    /**
+     * Default weights for organizing the view
+     */
+    private static final int[] DEFAULT_WEIGHTS_ShowQuery = new int[] { 450, 290, 260 };
+    private static final int[] DEFAULT_WEIGHTS_HideQuery = new int[] { 450, 100, 450 };
+    private static final int[] DEFAULT_WEIGHTS_FILTERING_H = new int[] { 3, 9 };
+    private static final int[] DEFAULT_WEIGHTS_TimeInterval = new int[] { 240, 380, 380 };
+
+    /** A composite that allows us to add margins for part A */
+    private @Nullable TmfXYChartViewer fChartViewerA;
+    private @Nullable SashForm fXYViewerContainerA;
+    private @Nullable TmfViewer fTmfViewerA;
+    private @Nullable SashForm fSashFormLeftChildA;
+    private @Nullable SashForm fSashFormLeftChildB;
+    private @Nullable IContextService fContextService;
+    private @Nullable Action fConfigureStatisticAction;
+    private List<IContextActivation> fActiveContexts = new ArrayList<>();
+
+    /** A composite that allows us to add margins for part B */
+    private @Nullable TmfXYChartViewer fChartViewerB;
+    private @Nullable SashForm fXYViewerContainerB;
+    private @Nullable TmfViewer fTmfViewerB;
+    private List<String> fTraceListA = new ArrayList<>();
+    private List<String> fTraceListB = new ArrayList<>();
+    private String fTraceStr = "Trace(s): ";//$NON-NLS-1$
```

Comment:
Should be static final and use Messages and Objects.requireNonNull. This comment should be applied for every static string in this file in a class attribute.